common/ffsparser.cpp

```diff
@@ -4937,13 +4928,27 @@ USTATUS FfsParser::parseCpdRegion(const UByteArray & region, const UINT32 localO
                 goto make_partition_table_consistent;
             }
             else {
-                msg(usprintf("%s: CPD partition can't fit into it's region, truncated", __FUNCTION__), partitions[i].index);
+                if (!partitions[i].hasMetaData && partitions[i].ptEntry.Offset.HuffmanCompressed) {
+                    msg(usprintf("%s: CPD partition is compressed but doesn't have metadata and can't fit into its region, length adjusted", __FUNCTION__),
+                        partitions[i].index);
+                }
+                else {
+                    msg(usprintf("%s: CPD partition can't fit into its region, truncated", __FUNCTION__), partitions[i].index);
+                }
                 partitions[i].ptEntry.Length = (UINT32)region.size() - (UINT32)partitions[i].ptEntry.Offset.Offset;
             }
         }
 
         // Check for intersection with previous partition
         if (partitions[i].ptEntry.Offset.Offset < previousPartitionEnd) {
+            // Check if previous partition was compressed but did not have metadata
+            if (!partitions[i - 1].hasMetaData && partitions[i - 1].ptEntry.Offset.HuffmanCompressed) {
```

Comment:
Is it safe to subtract 1 here?

Reply:
The index variable `i` starts at 1 in the for loop so there is always a `partitions[i - 1]` to compare with `partition[i]`.